static/js/components/email/EmailCompositionDialog.js

```diff
@@ -158,6 +205,7 @@ export default class EmailCompositionDialog extends React.Component {
       renderRecipients,
     } = this.props;
     const { editorState } = this.state;
+    let customButtons = R.map(this.makeCustomToolbarButton, ['PreferredName', 'Email']);
```

Comment:
here you could declare the `makeCustomToolbarButton` as an anonymous function wrapped with `R.map` above, which would avoid needing to declare this here (you could just inline the function call in the JSX below).